controllers/apps/transformer_component_deletion.go

```diff
@@ -143,6 +153,59 @@ func (t *componentDeletionTransformer) deleteCompResources(transCtx *componentTr
 	return graph.ErrPrematureStop
 }
 
+func handleRBACResourceDeletion(obj client.Object, transCtx *componentTransformContext, comp *appsv1.Component,
+	graphCli model.GraphClient, dag *graph.DAG) (skipDeletion bool, err error) {
+	switch v := obj.(type) {
+	case *corev1.ServiceAccount, *rbacv1.Role, *rbacv1.RoleBinding:
+		// list other components that reference the same componentdefinition
+		transCtx.Logger.V(1).Info("handling rbac resources deletion",
+			"comp", comp.Name, "name", klog.KObj(v).String())
+		compDefName := comp.Spec.CompDef
+		compList := &appsv1.ComponentList{}
+		if err := transCtx.Client.List(transCtx.Context, compList, client.InNamespace(comp.Namespace),
+			client.MatchingLabels{constant.ComponentDefinitionLabelKey: compDefName}); err != nil {
+			return false, err
```

Comment:
all component shared one serviceAccount in a namespace, even these are not belong to a same cluster?

Reply:
RoleBinding and serviceAccount are better managed by their respective components or sharding components

Reply:
> all component shared one serviceAccount in a namespace, even these are not belong to a same cluster?

Yes.

The idea is to reduce k8s objects. What are the cons of doing this?